Content.Server/GameObjects/Components/WiresComponent.cs

```diff
@@ -0,0 +1,243 @@
+            {
+                Guid = guid;
+                Identifier = identifier;
+                Color = color;
+                Owner = owner;
+                IsCut = isCut;
+            }
+        }
+
+        /// <summary>
+        /// Used by <see cref="IWires.RegisterWires"/>.
+        /// </summary>
+        public class WiresBuilder
+        {
+            [NotNull] private readonly WiresComponent _wires;
+            [NotNull] private readonly IWires _owner;
+
+            public WiresBuilder(WiresComponent wires, IWires owner)
+            {
+                _wires = wires;
+                _owner = owner;
+            }
+
+            public void CreateWire(object identifier, Color? color = null, bool isCut = false)
+            {
+                if (!color.HasValue)
+                {
+                    color = _wires.AssignColor();
+                }
+                else
+                {
+                    _wires._availableColors.Remove(color.Value);
+                }
+                _wires.WiresList.Add(new Wire(Guid.NewGuid(), identifier, color.Value, _owner, isCut));
+            }
+        }
+
+        /// <summary>
+        /// Picks a color from <see cref="_availableColors"/> and removes it from the list.
+        /// </summary>
+        /// <returns>The picked color.</returns>
+        private Color AssignColor()
+        {
+            if(_availableColors.Count == 0)
+            {
+                return Color.Black;
+            }
+            return _random.PickAndTake(_availableColors);
+        }
+
+        /// <summary>
+        /// Call this from other components to open the wires UI.
+        /// </summary>
+        public void OpenInterface(IPlayerSession session)
+        {
+            _userInterface.Open(session);
+        }
+
+        private void UserInterfaceOnReceiveMessage(ServerBoundUserInterfaceMessage serverMsg)
+        {
+            var message = serverMsg.Message;
+            switch (message)
+            {
+                case WiresActionMessage msg:
+                    var wire = WiresList.Find(x => x.Guid == msg.Guid);
+                    var player = serverMsg.Session.AttachedEntity;
+                    if (!player.TryGetComponent(out IHandsComponent handsComponent))
+                    {
+                        _notifyManager.PopupMessage(Owner.Transform.GridPosition, player, "You have no hands.");
+                        return;
+                    }
+                    var activeHandEntity = handsComponent.GetActiveHand?.Owner;
+                    switch (msg.Action)
+                    {
+                        case WiresAction.Cut:
+                            if (activeHandEntity?.HasComponent<WirecutterComponent>() != true)
+                            {
+                                _notifyManager.PopupMessage(Owner.Transform.GridPosition, player, "You need to hold a wirecutter in your hand!");
```

Comment:
Would be good to have some audio feedback here.

I know Eris has sounds, but the pulse sound is kinda obnoxious and doesn't fit well last I checked.

Reply:
It's also kind of annoying that the popups are layered under the UI, so if the window is open over the entity you might miss the message.
But re: audio, do you have any particular sound in mind? I don't know what would fit

Reply:
Honestly just a wirecutter sound and generic beep for pulsing would already be 10x better than nothing.

Reply:
Oh yeah, I forgot to mention this, but localize these strings too.

Reply:
Oh, I thought you were suggesting a sound for when the action fails because you left the comment on that line.